sound/soc/intel/boards/sof_sdw.c

```diff
@@ -984,17 +984,6 @@ static int create_sdw_dailink(struct snd_soc_card *card,
 	return 0;
 }
 
-/*
- * DAI link ID of SSP & DMIC & HDMI are based on last
- * link ID used by sdw link. Since be_id may be changed
- * in init func of sdw codec, it is not equal to be_id
- */
-static inline int get_next_be_id(struct snd_soc_dai_link *links,
```

Comment:
I'm not sure how this works @bardliao . This part of the code is already so confusing so I'm not saying its incorrect just a bit hard to follow. The DAI link ID is only set conditionally in create_sdw_link, right? WHat happens in case it isnt set ? Are you saying this function was unnecessary even before? In that case, this patch should precede the patch that sets a fixed DAI link ID right?

Reply:
ahh I see. makes sense.